recorder/recorder.go

```diff
@@ -145,6 +145,27 @@ func NewHook(handler HookFunc, kind HookKind) *Hook {
 // otherwise.
 type PassthroughFunc func(req *http.Request) bool
 
+// ErrUnsafeRequestMethod is returned when Options.BlockRealTransportUnsafeMethods is true, and
+// an request with an unsafe request is made.
+var ErrUnsafeRequestMethod = errors.New("request has unsafe method")
+
+type blockUnsafeMethodsRoundTripper struct {
+	RoundTripper http.RoundTripper
+}
+
+func (r *blockUnsafeMethodsRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	safeMethods := map[string]bool{
+		"GET":     true,
+		"HEAD":    true,
+		"OPTIONS": true,
+		"TRACE":   true,
```

Comment:
Please use the [constants from http package](https://pkg.go.dev/net/http#pkg-constants) here.